packages/angular/projects/ui-angular/src/lib/primitives-poc/button/button-component/button.component.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  HostBinding,
+  Input,
+  OnInit,
+  ViewEncapsulation,
+} from '@angular/core';
+
+@Component({
+  selector: 'button[amplify-button-poc]',
+  templateUrl: './button.component.html',
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AmplifyButtonComponent implements OnInit {
+  @Input() type: 'submit' | 'button' = 'button';
+  @Input() fullWidth: boolean | string = false;
+  @Input() isDisabled: boolean | string = false;
+  @Input() size: 'small' | 'medium' | 'large' = 'medium';
+  @Input() variation: 'primary' | 'default' | 'link' = 'default';
+  @Input() fontWeight: 'normal' | 'bold' | 'lighter' = 'normal';
+
+  @HostBinding('type') typeAttr: string;
+  @HostBinding('attr.data-fullwidth') fullWidthAttr: boolean | string;
+  @HostBinding('attr.data-size') sizeAttr: string;
+  @HostBinding('attr.data-variation') variationAttr: string;
+  @HostBinding('style.font-weight') fontWeightAttr: string;
+
+  @HostBinding('class') get classNames() {
+    let className = 'amplify-button';
+    const result = {
+      ...(this.variation && {
+        data: (className += ` amplify-button--${this.variation}`),
+      }),
+      ...(this.size && {
+        data: (className += ` amplify-button--${this.size}`),
+      }),
+      ...(this.fullWidth && {
+        data: (className += ` amplify-button--fullwidth`),
+      }),
+      ...(this.isDisabled && {
+        data: (className += ` amplify-button--disabled amplify-button--loading`),
+      }),
```

Comment:
Future improvement: we can use https://www.npmjs.com/package/classnames to simplify this a little bit. Not high priority!

Reply:
Implemented classname in component